cocotb/regression.py

```diff
@@ -39,6 +39,8 @@
 from itertools import product
 from typing import Any, Iterable, Optional, Tuple
 
+import pytest
```

Comment:
This would make `pytest` a hard dependency which, IIRC, it currently still isn't, right?

Reply:
I'll stick this in a try as well. We can default `Failed` to `AssertionError` in the case it isn't available.